src/NHibernate/Type/GenericSetType.cs

```diff
@@ -76,5 +77,20 @@ public override object Instantiate(int anticipatedSize)
 		{
 			return new HashSet<T>();
 		}
+
+		protected override bool AreCollectionElementsEqual(IEnumerable original, IEnumerable target)
```

Comment:
I am just a contributor as you, I will not decide if this is necessary. Those comments are just my thoughts.